### ISSUE TITLE
Add support for env variable

### ISSUE THREAD

**Issue by achachw** [Ouest-France/terraform-provider-ldap#37]
Hi there,
We are using this simple and very efficient provider. And we want to add support for getting bind_user and bind_password from environment variable.
#36 
What do you think about ?

**Comment by CorinneVerheyde** [comment]
Hi Walid, We already use environnement variables with this provider by defining terraform variables.
  bind_user     = var.ldap_user
  bind_password = var.ldap_password

And we export those variables this way :
  export TF_VAR_ldap_user='xxx'
  export TF_VAR_ldap_password='xxx'

**Comment by pablo-ruth** [comment]
Hi @achachw, is @CorinneVerheyde answer ok for you ?

**Comment by achachw** [comment]
Hi @CorinneVerheyde 
TF_VAR_ is a way to let terraform access  an env variable like a terraform variable.
My suggestion is to let the ldap provider getting credentials (username/password) directly from env variable. 
This is what aws provider do for example https://registry.terraform.io/providers/hashicorp/aws/latest/docs#environment-variables
also many others provider do (vsphere, azurerm, ..)
With this uses can set their ldap provider like this : 
```
provider "ldap" {
  host          = "myhost"
  port          = 389
}
```

We can even export host and port, so provider will be configured simply like this : 
```
provider "ldap" {}
```

**Comment by pablo-ruth** [comment]
I think that env vars for AWS and other cloud providers are more a compatibility choice with their CLI than a choice for their Terraform providers. I'm not against adding it to this provider if it's easier to use in some cases. I think adding a "LDAP_" prefix to those env vars can make them clearer, is it right for you ?

**Comment by CorinneVerheyde** [comment]
Yes, indeed it can be easier in some cases.
+1 for adding "LDAP_" prefix.

**Comment by achachw** [comment]
Go for LDAP_USER, LDAP_PASSWORD.
i am going to update this PR suggestion #36

**Comment by achachw** [comment]
Done. 
Please let me know if you want me to make the same for LDAP_PORT and LDAP_HOST.
Regards

**Comment by pablo-ruth** [comment]
Thanks, yes it would be great to have the same for the host and port, it would be more coherent. Can you please update the docs too?

I'll check this tomorrow.

**Comment by achachw** [comment]
Hi @pablo-ruth 
I updated docs and add support for host and port.
Could you please have a look on my PR and see if it's content meet you expectation ?
Thanks

**Comment by pablo-ruth** [comment]
Hi @achachw, LGTM. Thanks for your contribution! I merged it and released a [0.7.3](https://github.com/Ouest-France/terraform-provider-ldap/releases/tag/v0.7.3) versions.
